syft/frameworks/torch/nn/functional.py

```diff
@@ -150,3 +150,56 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1):
         .contiguous()
     )
     return res
+
+
+def _pool(tensor, kernel_size: int = 2, stride: int = 2, mode="max"):
+    output_shape = (
+        (tensor.shape[0] - kernel_size) // stride + 1,
+        (tensor.shape[1] - kernel_size) // stride + 1,
+    )
+    kernel_size = (kernel_size, kernel_size)
+    b = torch.ones(tensor.shape)  # when torch.Tensor.stride() is supported: replace with A.stride()
+    a_strides = b.stride()
+    a_w = torch.as_strided(
+        tensor,
+        size=output_shape + kernel_size,
+        stride=(stride * a_strides[0], stride * a_strides[1]) + a_strides,
+    )
+    a_w = a_w.reshape(-1, *kernel_size)
+    result = []
+    if mode is "max":
+        for channel in range(a_w.shape[0]):
+            result.append(a_w[channel].max())
+    else:
+        for channel in range(a_w.shape[0]):
+            result.append(torch.mean(a_w[channel]))
+
+    result = torch.stack(result).reshape(output_shape)
+    return result
+
+
+def pool2d(tensor, kernel_size: int = 2, stride: int = 2, mode="max"):
+    assert len(tensor.shape) == 4
```

Comment:
`torch.nn.functional.max_pool2d` currently supports both 3D and 4D tensors. So might be good to add compatibility for 3D tensors too.

Reply:
that's an initial version to get it to work then we'll iterate on it

Reply:
Can we have this done before we merge? I guess it might be work if we have something like 
```python
if len(tensor.shape) == 3:
    return squeeze(pool2d(unsqueeze(tensor))
```
or something like that??